src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -1,138 +1,433 @@
-      onChange: ListboxOnChange;
-      // These are props that are never allowed on the listbox type.
-      onMixedStateChange?: never;
-    }
-  | {
-      type: "dialog";
-      /** The action to perform for save/apply button of multiselect. */
-      onApply: () => void;
-      /** The action to perform on the checkbox's onChange function.  */
-      onChange: DialogOnChange;
-      /** The action to perform for a mixed state checkbox (parent checkbox). */
-      onMixedStateChange?: DialogOnChange;
-    };
-
-export type MultiSelectProps = MultiSelectCommonProps &
-  MultiSelectVariantsProps;
+};
 
 /**
  * The `MultiSelect` component is a form input element that presents a list
  * of `Checkbox` components from which a user can make one or multiple
- * selections. Two variants of the MultiSelect component are offered, each with
- * slightly different functionality and requirements.  Because of these
- * differences, the two variants are broken out in separate stories below.
+ * selections. The MultiSelect allows for an optional set of child checkboxes to be passed,
+ * which makes the “parent” checkbox function as a check/uncheck all toggle.
+ * If all of the children checkboxes are checked, the parent isChecked prop will be true.
+ * If only some of the child checkboxes are checked, the parent checkbox will have a isIndeterminate state,
+ * that can be checked or unchecked.
  */
+
 export const MultiSelect = chakra(
   forwardRef<HTMLDivElement, React.PropsWithChildren<MultiSelectProps>>(
-    (props, ref?) => {
+    (props, _ref?) => {
       const {
+        helperText,
         id,
         isBlockElement = false,
         isDefaultOpen = false,
+        isSearchable = false,
+        defaultItemsVisible = 5,
         items,
-        labelText,
-        onApply,
+        listOverflow = "scroll",
+        buttonText,
         onChange,
         onClear,
         onMixedStateChange,
         selectedItems,
-        type,
-        width = "default",
+        width = "full",
         ...rest
       } = props;
 
-      const commonProps = {
-        id,
-        isBlockElement,
-        isDefaultOpen,
-        items,
-        labelText,
-        onClear,
-        selectedItems,
-        type,
+      // Control the open or closed state of the MultiSelect.
+      const [isOpen] = useState(isDefaultOpen);
+      const [itemsList, setItemsList] = useState(items);
+      const [viewAllLabel, setViewAllLabel] = useState("View all");
+      const [listHeight, setListHeight] = useState("215px");
+      const [listItemsCount, setListItemsCount] = useState(defaultItemsVisible);
+
+      // Separate effect for handling listOverflow "scroll"
+      React.useEffect(() => {
+        if (listOverflow === "scroll") {
+          setListHeight("215px");
+          if (isSearchable || items.some((item) => !!item.children)) {
+            setListHeight("275px");
+          }
+          setItemsList(items);
+        }
+      }, [listOverflow, items]);
```

Comment:
Github and the linter tells you that you need additional dependencies in the array:
<img width="970" alt="Screen Shot 2024-01-18 at 6 32 49 PM" src="https://github.com/NYPL/nypl-design-system/assets/1280564/c668c976-ef80-474f-93b3-3aa5cb933749">